### ISSUE TITLE
Overwrite mode not overwriting partitions in S3 with `viaParquet`?

### ISSUE THREAD

**Issue by jhenahan** [mjakubowski84/parquet4s#289]
I've got a pipe at the end of my pipeline like this:

```scala
object WriteMachinery {

  private val config: Configuration = {
    val hadoop = new Configuration()
    hadoop.set("fs.s3a.path.style.access", "true")
    hadoop.set("fs.s3a.endpoint.region", Region.US_EAST_1.id())
    hadoop
  }

  def writePipe(
      prefix: String
  ): Pipe[IO, MyType, MyType] = {
    viaParquet[IO]
      .of[MyType]
      .options(
        ParquetWriter.Options(
          writeMode = ParquetFileWriter.Mode.OVERWRITE,
          compressionCodecName = CompressionCodecName.SNAPPY,
          hadoopConf = config
        )
      )
      .partitionBy(Col("col1"), Col("col2"), Col("col3"))
      .write(Path(prefix))
  }
}
```

This does properly write data with Hive-style partitions at `s3://prefix/col1=*/col2=*/col3=*` when I send data through the pipe. My expectation is that if I run the same data through this pipe a second time (or, for that matter, some different data), the entire prefix should be emptied out. What am I missing?

EDIT: Parquet-related deps:

```scala
lazy val parquet = Seq(
  "com.github.mjakubowski84" %% "parquet4s-core" % "2.8.0",
  "com.github.mjakubowski84" %% "parquet4s-fs2"  % "2.8.0",
  "org.apache.hadoop"         % "hadoop-client"  % "3.3.4",
  "org.apache.hadoop" % "hadoop-aws" % "3.3.4" exclude ("com.amazonaws", "aws-java-sdk-bundle"),
  "com.amazonaws" % "aws-java-sdk-s3"       % "1.12.376",
  "com.amazonaws" % "aws-java-sdk-core"     % "1.12.376",
  "com.amazonaws" % "aws-java-sdk-dynamodb" % "1.12.376"
)
```

I'm also trying a version where I don't exclude the SDK bundle in case there's something amiss there 🙃

**Comment by mjakubowski84** [comment]
Hi @jhenahan,

I am not sure what's going on. Parquet4s, by design, deletes the whole directory indicated in `.write(Path(prefix))` when override mode is in use.
I recommend turning on debug logs. Maybe there's an issue with permissions?

**Comment by jhenahan** [comment]
```diff
diff --git i/fs2/src/main/scala/com/github/mjakubowski84/parquet4s/parquet/rotatingWriter.scala w/fs2/src/main/scala/com/github/mjakubowski84/parquet4s/parquet/rotatingWriter.scala
index a60f59a..0baadcb 100644
--- i/fs2/src/main/scala/com/github/mjakubowski84/parquet4s/parquet/rotatingWriter.scala
+++ w/fs2/src/main/scala/com/github/mjakubowski84/parquet4s/parquet/rotatingWriter.scala
@@ -517,6 +517,7 @@ object rotatingWriter {
         logger     <- Stream.eval(logger[F](this.getClass))
         eventQueue <- Stream.eval(Queue.unbounded[F, WriterEvent[F, T, W]])
         writersRef <- Stream.eval(Ref.of(Map.empty[Path, RecordWriter[F]]))
+        _          <- Stream.eval(io.validateWritePath[F](basePath, options, logger))
         rotatingWriter <- Stream.emit(
           new RotatingWriter[T, W, F](
             basePath            = basePath,
```

This is my local diff. Initial testing looks promising. :)

**Comment by mjakubowski84** [comment]
You are right! It turns out that overwrite mode has never been implemented in `viaParquet`. And to implement it properly, `validateWritePath` has to be modified, too.

**Comment by mjakubowski84** [comment]
The fix released is released in v2.9.0
